paddle/framework/framework.proto

```diff
@@ -118,6 +118,38 @@ message LoDTensorArrayDesc {
 
 message ReaderDesc { repeated LoDTensorDesc lod_tensor = 1; }
 
+message MetaDataDesc {
+  enum DataType {
+    BOOL = 1;
+    INT16 = 2;
+    INT32 = 3;
+    INT64 = 4;
+    FP16 = 5;
+    FP32 = 6;
+    FP64 = 7;
+    TENSOR = 8;
+    LOD_TENSOR = 9;
+    SELECTED_ROWS = 10;
+    LOD_TENSOR_ARRAY = 11;
+  }
+  required DataType base_type = 1;
+  optional bool data_bool = 2;
+  optional int32 data_int32 = 3;
+  optional int64 data_int64 = 4;
+  optional float data_float = 5;
+  optional LoDTensorDesc lod_tensor = 6;
+  optional TensorDesc selected_rows = 7;
+  optional LoDTensorArrayDesc tensor_array = 8;
+}
+
+message ChanEleDesc { repeated MetaDataDesc meta_data = 1; }
```

Comment:
I assume this step would append multiple channel elements (like send receive etc.) together or does it stack multiple items of the same element together (like all the send requests and all the receive requests).

Reply:
We can regard the `ChanEleDesc`  as a tuple, it can describe <int32,int32,int64,float32>.